### ISSUE TITLE
Make openai and other things optional dependencies

### ISSUE THREAD

**Issue by prabhu** [bridgecrewio/checkov#5063]
Many, including me, aren't fans of piping sensitive code to Microsoft via openai. Could these things become optional?

The ideal workflow must be

```
pip install checkov
# Must only install a simple IaC linting tool
```

```
pip install checkov[openai]
# Provide disclaimer and instructions to enable the integration
```

https://github.com/bridgecrewio/checkov/blob/main/Pipfile#LL89-L89C7

**Comment by gruebel** [comment]
hey @prabhu thanks for reaching out.

We only leverage OpenAI, when the flag `--openai-api-key` is used, otherwise nothing is send to it. We try to offer as much possibilities as possible and make it easy to leverage all features, that's why we currently don't plan on offering a subset of the features, because you can control via env vars or flags what to use or not.

**Comment by DevJake** [comment]
> We only leverage OpenAI, when the flag `--openai-api-key` is used, otherwise nothing is send to it. We try to offer as much possibilities as possible and make it easy to leverage all features, that's why we currently don't plan on offering a subset of the features, because you can control via env vars or flags what to use or not.

@gruebel I understand this sentiment for the project. However, I am working on a large project that uses `openai` already. This library depends on `openai<1.0.0`, whereas their current version is `1.35.14`. In effect, I cannot use Checkov because of the version mismatches. I do not use the OpenAI integration that Checkov offers through `--openai-api-key`. I would suggest a fix to be any of:
1. Remove the dependency version lock of `openai`
2. Keep the library's dependencies up to date with the latest versions of `openai`
3. Make the OpenAI integration optional, as this issue originally requested

Any of these would fix the issue, although #1 and #3 are most preferable. Let me know your thoughts!
